macros/qualtrics_union_data.sql

```diff
@@ -0,0 +1,137 @@
+{%- macro qualtrics_union_data(table_identifier, database_variable, schema_variable, default_database, default_schema, default_variable, union_schema_variable='union_schemas', union_database_variable='union_databases') -%}
```

Comment:
Since we're already making edits to customize the union data macro, what are your thoughts on taking this opportunity to leverage the newer solution like the one implemented in [Zendesk](https://github.com/fivetran/dbt_zendesk_source/tree/main/macros/union)?

Especially since I notice there is some leftover code in this macro that doesn't pertain to this package (like the need to adjust the identifier for the Twitter Ads package). It may be more appropriate in this case to overhaul this system entirely. This way if we need to implement in other packages then we have a reference to fall back on.

Reply:
That is a very good point to consider.

The biggest thing is that if we were to switch to the Zendesk-esque solution, this would be a breaking change for non-Quickstart users who are currently using the unioning feature, as we'd switch from requiring a list to a dictionary. 

However, I think we could make this backwards compatible (though the code to do so might be a lil complex). We could use the new macro if the new dictionary variable is found (else use the old way), or we could cobble together a dictionary out of the `qualtrics_union_schemas/databases` lists if those are provided.

Reply:
Discussed offline and we'll keep the old way for now (with the Redshift update) and circle back in 2 months to apply the same update to the fivetran_utils macro